### ISSUE TITLE
please dont change the API as you again did in latest release. 

### ISSUE THREAD

**Issue by fduncanh** [libimobiledevice/libplist#260]
Thanks for your great work, but....

>Changes
> Change API around PLIST_DATA to use uint8_t* instead of char*

Changing the API is disruptive, and should not be done on a whim to make things "cleaner"

Better to introduce a new function, and replace the previous one with a wrapper to the new version.  We have to support all versions of libplist later than some quite old one   (v.2.0)

* please consider releasing v 2.5.0.1  to revert this change to plist_get_data_val, and rename the modified one to plist_get_data_val_u8  (etc.  for the other changed functions)  It's only a few days since you released this disruptive change, it's not too late to backtrack.

I suggest:

```
PLIST_API void plist_get_data_val_u8(plist_t node, uint8_t **val, uint64_t * length);

// deprecated: do not use in new code:
PLIST_API void plist_get_data_val(plist_t node, char **val, uint64_t * length) 
{
    void plist_get_data_val_u8(node, (uint8_t ** ) val, * length);
}
```
(+ similar changes for plist_set_data_val, plist_get_data_ptr, plist_new_data )

* **There is no point in gratutiously breaking things for your users!**  

e.g.: 

https://github.com/FDH2/UxPlay/issues/294

**Comment by fduncanh** [comment]
@nikias 
in case you are willing to restore compatibility with the older API, I have provided a pull request #261 to do this.

**Comment by fduncanh** [comment]
I've now worked around this in the code I maintain with cmake to identify the libplist version, and then #define appropriate casts when the version >= 2.5,  but it would still be a good idea to you to provide the old API functions as wrappers around renamed uint8_t functions as in PR #261.   Sorry if I was grumpy.

**Comment by nikias** [comment]
Sorry guys I reverted the changed and released a 2.6.0

**Comment by fduncanh** [comment]
THANK YOU!!!
